src/main/java/org/jlab/mya/nexus/DatabaseSourceStream.java

```diff
@@ -103,14 +104,30 @@ public boolean isOpen() {
     }
 
     /**
-     * Closes the channel.
+     * Closes the channel and underlying database resources.
      *
      * @throws IOException If an I/O error occurs
      */
     @Override
     public void close() throws IOException {
         try {
-            con.close(); // TODO: Make sure closing connection also closes stmt and rs (pooled wrappers I'm looking at you)
```

Comment:
Glad we finally got around to this TODO!  A quick google reveals PooledConnections are indeed the exception and do not always clean up like you'd hope/expect (might be implementation dependent).  Non-pooled connections do automatically clean up.